### PR TITLE
XWIKI-17072: Warnings for deprecated usage of HQL ordinal parameters …

### PR DESCRIPTION
**Pull request by oanalavinia**
…when applying a date filter on Deleted Attachments livetable

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedAttachments.xml [hunk 1]
```diff
@@ -156,13 +156,13 @@ $response.setContentType('application/json')
   #set($dates = $dateFilter.split('-'))
   #if($dates.size() == 2)
     ## Date range matching
-    #set($sql = "${sql} and datt.date between ? and ? ")
-    #set($discard = $params.add($datetool.toDate($numbertool.toNumber($dates[0]))))
-    #set($discard = $params.add($datetool.toDate($numbertool.toNumber($dates[1]))))
+    #set($sql = "${sql} and datt.date between :start_date and :end_date ")
+    #set($discard = $params.put('start_date', $datetool.toDate($numbertool.toNumber($dates[0]))))
+    #set($discard = $params.put('end_date', $datetool.toDate($numbertool.toNumber($dates[1]))))
```
Comment (mflorea) [inline xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-ui/src/main/resources/XWiki/DeletedAttachments.xml]: I would use camel case for the parameter names, to be in-line with our code style.

**Comment by vmassol** [comment]
@oanalavinia you might want to give a go at https://jira.xwiki.org/browse/XWIKI-17072?focusedCommentId=104843&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-104843 too ;)

**Comment by oanalavinia** [comment]
> @oanalavinia you might want to give a go at https://jira.xwiki.org/browse/XWIKI-17072?focusedCommentId=104843&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-104843 too ;)

I'll give it a try :)

**Comment by mflorea** [comment]
@vmassol @oanalavinia will try to find some time to write a functional test for this before 12.2RC1. She will make a separate PR.

**Comment by vmassol** [comment]
> @vmassol @oanalavinia will try to find some time to write a functional test for this before 12.2RC1. She will make a separate PR.

TBH I don't know how hard it is. If too hard then forget it or ask for help. Thanks! That would be awesome and improve our quality :)

**Pull request merged**